### PR TITLE
Improved performance drawHorizontalBezier

### PR DESCRIPTION
**Pull request by Banannzza**
### Issue Link :link:
https://github.com/danielgindi/Charts/issues/3550

### Goals :soccer:
Reduce execution time of drawHorizontalBezier

### Implementation Details :construction:
Explanation of the decision on the link: https://stackoverflow.com/a/6353597

### CODE REVIEW

**Comment by liuxuan30** [comment]
hi there, thanks for investigating on this. Have you tested on your side? Can you post the comparison results?

File: Source/Charts/Renderers/LineChartRenderer.swift [hunk 1]
```diff
@@ -246,6 +246,10 @@ open class LineChartRenderer: LineRadarRenderer
         context.beginPath()
         context.addPath(cubicPath)
         context.setStrokeColor(drawingColor.cgColor)
+        context.setLineCap(CGLineCap.butt)
```
Comment (liuxuan30) [inline Source/Charts/Renderers/LineChartRenderer.swift]: we have a var for line cap
```
    /// Line cap type, default is CGLineCap.Butt
    open var lineCapType = CGLineCap.butt
```
in line chart data set.

File: Source/Charts/Renderers/LineChartRenderer.swift [hunk 2]
```diff
@@ -246,6 +246,10 @@ open class LineChartRenderer: LineRadarRenderer
...
+        context.setLineJoin(CGLineJoin.round)
```
Comment (liuxuan30) [inline Source/Charts/Renderers/LineChartRenderer.swift]: If configuring line cap and line join, miter limit, flatness will improve the performance, please create a new var for each of them, and maybe help to test the performance on your side.

I'm not sure if it has to be hardcoded params.

**Comment by liuxuan30** [comment]
@petester42 @jjatie @danielgindi are you aware of this subtle trick?

**Comment by petester42** [comment]
Nope

**Comment by liuxuan30** [comment]
I have given some thoughts on this, I'm not sure if setting those properties could improve our performance. The difference is the context of our `context`.

@Banannzza can you post the performance data, e.g. instruments time cost before and after you apply these changes?

**Comment by jjatie** [comment]
@liuxuan30 @Banannzza I read the SO post differently. Why are we not just using a `UIBezierPath`? If even in their best implementations they are equal in performance, we should be using the higher level API. If Apple is able to make further optimizations, `UIBezierPath` will get it for free.

**Comment by petester42** [comment]
Sure it would be a better API for adding points to draw but it won’t address the core performance issue. The issue is that we are using core graphics uses too much cpu when drawing.

**Comment by jjatie** [comment]
@petester42 Since we're only supporting iOS 8+, shouldn't these CG calls be hitting the GPU more than the CPU? Since Metal was introduced Core Graphics is supposed to be backed by Metal. I've not looked into this too much.

**Comment by petester42** [comment]
No idea. Historically it has not been the case. I believe the actual drawing is done on the gpu but all the setup code is not.

**Comment by liuxuan30** [comment]
what's the status of this? Should we merge or hold?

**Pull request closed without merging**